### ISSUE TITLE
i18nFormat returns null after upgrading to 3.2

### ISSUE THREAD

**Issue by BassemN** [cakephp/cakephp#8142]
Hi,

After upgrading to 3.2, `$this->Time->i18nFormat` returns null into my views as you can see on the below code

`<?= $this->Time->i18nFormat( $post->created, null, null, 'America/Chicago' ); ?>`

Thanks

**Comment by markstory** [comment]
I wasn't able to reproduce this issue within a new sample application. What is the output of `var_dump($post->created)`?

**Comment by BassemN** [comment]
Kindly check the below output for `var_dump($post->created)`

`object(Cake\I18n\FrozenTime)#187 (3) { ["date"]=> string(26) "2015-12-23 09:48:36.000000" ["timezone_type"]=> int(3) ["timezone"]=> string(3) "UTC" }`

**Comment by antograssiot** [comment]
@BassemN what version of php are you using ?

**Comment by BassemN** [comment]
I have tested it into two PHP versions 5.5.12 & 5.5.31

**Comment by antograssiot** [comment]
@BassemN the following test is passing for me, could you test it also ?

```php
$time = new FrozenTime('2015-12-23 09:48:36.000000');
$result = $this->Time->format($time, null, null, 'America/Chicago');
$expected = '12/23/15 3:48 AM';
$this->assertTimeFormat($expected, $result);
```

**Comment by BassemN** [comment]
I got the reason of that issue. If I changed the locale at runtime by adding the below code into bootstrap.php file, `i18nFormat` or `format` returns null but If I removed that code it returns the date
`\Cake\I18n\I18n::locale( 'de_DE' );`

**Comment by antograssiot** [comment]
@BassemN could you try adding this to your bootstrap if it's not already present.

```php
Type::build('date')->useLocaleParser();
Type::build('datetime')->useLocaleParser();
```

**Comment by BassemN** [comment]
@antograssiot It is already there and I added it again and it is still not working

**Comment by antograssiot** [comment]
@BassemN what do you get when you use 

```php
$result = $this->Time->format($time, null, false, 'America/Chicago');
```
It might help to catch the exception you are getting

**Comment by BassemN** [comment]
@antograssiot  check the below error log
```
2016-01-31 14:15:29 Error: [Exception] DateTime::__construct(): 
Request URL: /cs/
Stack Trace:
#0 /var/www/html/cs/vendor/cakephp/chronos/src/MutableDateTime.php(60): DateTime->__construct('\xD9\xA2\xD9\xA3\xE2\x80\x8F/\xD9\xA1\xD9\xA2\xE2\x80\x8F...', NULL)
#1 /var/www/html/cs/vendor/cakephp/cakephp/src/I18n/Time.php(114): Cake\Chronos\MutableDateTime->__construct(Object(Cake\I18n\FrozenTime), NULL)
#2 /var/www/html/cs/vendor/cakephp/cakephp/src/View/Helper/TimeHelper.php(337): Cake\I18n\Time->__construct(Object(Cake\I18n\FrozenTime))
#3 /var/www/html/cs/vendor/cakephp/cakephp/src/View/Helper/TimeHelper.php(315): Cake\View\Helper\TimeHelper->i18nFormat(Object(Cake\I18n\FrozenTime), NULL, false, 'America/Chicago')
#4 /var/www/html/cs/src/Template/Posts/index.ctp(22): Cake\View\Helper\TimeHelper->format(Object(Cake\I18n\FrozenTime), NULL, false, 'America/Chicago')
#5 /var/www/html/cs/vendor/cakephp/cakephp/src/View/View.php(992): include('/var/www/html/c...')
#6 /var/www/html/cs/vendor/cakephp/cakephp/src/View/View.php(952): Cake\View\View->_evaluate('/var/www/html/c...', Array)
#7 /var/www/html/cs/vendor/cakephp/cakephp/src/View/View.php(587): Cake\View\View->_render('/var/www/html/c...')
#8 /var/www/html/cs/vendor/cakephp/cakephp/src/Controller/Controller.php(612): Cake\View\View->render(NULL, NULL)
#9 /var/www/html/cs/vendor/cakephp/cakephp/src/Routing/Dispatcher.php(120): Cake\Controller\Controller->render()
#10 /var/www/html/cs/vendor/cakephp/cakephp/src/Routing/Dispatcher.php(87): Cake\Routing\Dispatcher->_invoke(Object(App\Controller\PostsController))
#11 /var/www/html/cs/webroot/index.php(37): Cake\Routing\Dispatcher->dispatch(Object(Cake\Network\Request), Object(Cake\Network\Response))
#12 {main}
```

**Comment by markstory** [comment]
Ok, I was able to reproduce the issue by setting my default locale to de_DE as well. I'll take a look at why this is failing later today.

**Comment by antograssiot** [comment]
@markstory it seems that the problem comes from converting DateTimeImmutable to DateTime .
In a view, the following would fail
```php 
\Cake\I18n\I18n::locale('de_DE');
$time = new \Cake\I18n\FrozenTime('2015-12-23 09:48:36');
echo $this->Time->format($time);
```
and this would work
```php 
\Cake\I18n\I18n::locale('de_DE');
$time = new \Cake\I18n\FrozenTime('2015-12-23 09:48:36');
$time = $time->toMutable();
echo $this->Time->format($time);
```

**Comment by dereuromark** [comment]
Sounds like the Time helper itself is not "read only", but tries to modify the Time object.

**Comment by markstory** [comment]
Fixed in #8146
